### PR TITLE
Unittests for a couple of functions

### PR DESCRIPTION
**Pull request by siong-chin**

### CODE REVIEW

File: main/githooks.py [hunk 1]
```diff
@@ -306,7 +306,17 @@ def _test(input, output=None):
 
 
 def trim_trailing_whitespace_in_file(filename, new_file, in_place):
-    '''Remove trailing white spaces in new and modified lines in a filename'''
+    '''Remove trailing white spaces in new and modified lines in a filename
+
+    :param filename: The file to check
+    :param new_file: True if whole file is new; False if it's an existing
+        file that's been modified
+    :param in_place: True if trailing whitespace is to be removed from file in
```
Comment (rockdreamer) [inline main/githooks.py]: I find the wording of the variable doesn't help me understand what's happening. I've seen dry_run used as an expression for this. You could also call it do_not_change_files

File: main/githooks.py [hunk 2]
```diff
@@ -350,6 +360,27 @@ def trim_trailing_whitespace_in_file(filename, new_file, in_place):
     return 0
 
 
+class TestTrimTrailingWhitespace(unittest.TestCase):
+    def test_trim_trailing_whitespace(self):
+        test_file = Path.cwd() / 'tttw_testfile.txt'
```
Comment (rockdreamer) [inline main/githooks.py]: cwd might not be what you expect as I've been finding out lately :)
Could you use a context manager from tempfile please?

File: main/githooks.py [hunk 3]
```diff
@@ -411,6 +442,20 @@ def check_filename(filepath):
...
+class TestCheckFileName(unittest.TestCase):
+    def test_various_strings(self):
+        def _test(input, output):
+            self.assertEqual(output, check_filename(input))
+        _test('good/some.txt', 0)
+        _test('bad/illegal/star*star.txt', 1)
+        _test('bad/reserved/device/con.txt', 1)
+        _test('bad/end/period.txt.', 1)
+        _test('bad/end/space.txt ', 1)
+        _test('bad/ascii/你好.txt', 1)
```
Comment (rockdreamer) [inline main/githooks.py:503]: Sigh: proper handling of unicode is still far from a solved problem

**Review by rockdreamer — APPROVED** [review]

File: main/githooks.py (hunk 1 above)
Comment (siong-chin) [reply main/githooks.py]: Will change to dry_run.

File: main/githooks.py (hunk 2 above)
Comment (siong-chin) [reply main/githooks.py]: Good point. Done.

File: main/githooks.py (hunk 3 above)
Comment (siong-chin) [reply main/githooks.py:503]: This is a good check. In my last workplace we had some test files with non ascii file names and some systems handled them in unexpected way.

**Pull request merged**